solutions/ojuz-knapsack.mdx

```diff
@@ -0,0 +1,171 @@
+---
+id: ojuz-knapsack
+source: Singapore NOI
+title: knapsack
+author: Kevin Sheng
+---
+
+## Explanation
+
+Note that the bound on $S$ is rather small and that $1 \leq W_i \leq S$, so we can group the items together by weight.  
+After this, we can implement the knapsack problem on the grouped together items. We also sort items of the same weight by reverse value, as it is always optimal to select the item with greater value among two items of the same weight.  
+
+## Implemention
+
+**Time Complexity:** $\mathcal{O}(S^2 \cdot \log S)$
```

Comment:
Add an explanation to this time complexity plz

Reply:
jokes on you i don't understand it either

Reply:
ok i just copy pasted the thing in there, i'm going to trust you